go/vt/mysqlctl/capabilities.go

```diff
@@ -0,0 +1,248 @@
+	default:
+		log.Warningf("%d: unknown capability", capability)
+		return false
+	}
+
+}
+
+func (mysqld *Mysqld) greaterThan(version serverVersion) bool {
+	if mysqld.version[0] > version[0] {
+		return true
+	} else if mysqld.version[0] == version[0] && mysqld.version[1] > version[1] {
+		return true
+	} else if mysqld.version[0] == version[0] && mysqld.version[1] == version[1] && mysqld.version[2] >= version[2] {
+		return true
+	} else {
+		return false
+	}
+}
+
+// IsMySQLLike tests if the server is either MySQL
+// or Percona Server. At least currently, Vitess doesn't
+// make use of any specific Percona Server features.
+func (mysqld *Mysqld) IsMySQLLike() bool {
+	return mysqld.flavor == flavorMySQL || mysqld.flavor == flavorPercona
+}
+
+// IsMariaDB tests if the server is MariaDB.
+// IsMySQLLike() and IsMariaDB() are mutually exclusive
+func (mysqld *Mysqld) IsMariaDB() bool {
+	return mysqld.flavor == flavorMariaDB
+}
+
+// MajorVersion returns an integer for the leading version.
+// For example 8 in the case of MySQL 8.0.11, or
+// 5 in the case of 5.6.15
+func (mysqld *Mysqld) MajorVersion() int {
+	return mysqld.version[0]
+}
+
+// MinorVersion returns an integer for the second version
+// For example 6 in the case of 5.6.11
+func (mysqld *Mysqld) MinorVersion() int {
+	return mysqld.version[1]
+}
+
+// PatchVersion returns an integer for the third
+// part of the version. For example 11 in the case of
+// 5.6.11
+func (mysqld *Mysqld) PatchVersion() int {
+	return mysqld.version[2]
+}
+
+func (mysqld *Mysqld) getVersionString() (string, error) {
+
+	mysqlRoot, err := vtenv.VtMysqlRoot()
+	if err != nil {
+		return "", err
+	}
+	mysqldPath, err := binaryPath(mysqlRoot, "mysqld")
+	if err != nil {
+		return "", err
+	}
+
+	_, version, err := execCmd(mysqldPath, []string{"--version"}, nil, mysqlRoot, nil)
+	if err != nil {
+		return "", err
+	}
+
+	return version, nil
+
+}
+
+func (mysqld *Mysqld) detectFlavor() (string, error) {
+
+	mysqlFlavor := os.Getenv("MYSQL_FLAVOR")
+	if len(mysqlFlavor) > 0 {
+		mysqlFlavor = strings.ToLower(mysqlFlavor)[0:5]
+
+		if mysqlFlavor == "mysql" {
+			return flavorMySQL, nil
+		} else if mysqlFlavor == "maria" {
+			return flavorMariaDB, nil
+		}
+	}
+
+	// Flavor was either not defined in ENV, or unknown
+
+	version, err := mysqld.getVersionString()
+	if err != nil {
+		return version, err
+	}
+
+	// Check if the flavor has been defined
+	// This takes first precedence.
+
+	if strings.Contains(version, "MySQL Community Server") {
+		return flavorMySQL, nil
+	} else if strings.Contains(version, "Percona") {
+		return flavorPercona, nil
+	} else if strings.Contains(version, "MariaDB") {
+		return flavorMariaDB, nil
+	}
+
+	return "", fmt.Errorf("Could not autodetect flavor!")
+
+}
+
+func (mysqld *Mysqld) detectVersion() (serverVersion, error) {
+
+	mysqlFlavor := strings.ToLower(os.Getenv("MYSQL_FLAVOR"))
+
+	// Try not to be too smart with version parsing.
+	// Assume GA versions
```

Comment:
I've removed it instead. I think because MySQL 8.0 has a lot of features in patch releases, it may be hard to maintain an environment variable that only includes major/minor.